acceptance/tests/ensure_puppet_facts_works_with_facter_ng.rb

```diff
@@ -0,0 +1,19 @@
+test_name 'PA-3114: Use facter-ng on all platforms' do
+
+  require 'puppet/acceptance/common_utils'
+
+  teardown do
+    on agent, puppet('config set facterng false')
+  end
+
+  agents.each do |agent|
+    step "test puppet facts with facter-ng'" do
+      on agent, puppet('config set facterng true')
+      on agent, puppet('facts'), :acceptable_exit_codes => [0] do
+        facter_major_version = stdout.chomp.split("\n").select{|line|  line.match?(/facterversion/)}
+        assert_match(/4.[0-9]+.[0-9]+/, facter_major_version.first, "puppet failed to change cfacter to facter-ng")
```

Comment:
```
facter_major_version = stdout.split("\n").find { |line|  line.match?(/facterversion/) }
assert_match(/4.[0-9]+.[0-9]+/, facter_major_version, "puppet failed to change cfacter to facter-ng")
```

`find` will stop at the first occurrence. Alternatively you could `JSON.prase(stdout)` and access the `facterversion` key